[Core] Update PG doc to clarify the behavior of scheduling actor with default resource requirement (#45778)
ray-project/ray: doc/source/ray-core/scheduling/placement-group.rst

.. note::

>   By default, Ray actors use 1 logical CPU for scheduling, and 0 logical CPU for running.
>   This means, by default, actors cannot get scheduled on a zero-cpu node, but an infinite number of them can run on any non-zero cpu node.
>   The default resource requirements for actors was chosen for historical reasons.
>   When scheduling an actor with default resource requirements and a placement group, the placement group has to be created with a bundle containing at least 1 CPU
>   (since the actor requires 1 CPU for scheduling). However, after the actor is created, it doesn't consume any placement group resources.

Reviewer: ```suggestion
  By default, Ray actors require 1 logical CPU to be available at schedule time, but after being scheduled, they do not acquire any CPU resources.
  In other words, by default, actors cannot get scheduled on a zero-cpu node, but an infinite number of them can run on any non-zero cpu node.

  Thus, when scheduling an actor with the default resource requirements and a placement group, the placement group has to be created with a bundle containing at least 1 CPU
  (since the actor requires 1 CPU for scheduling). However, after the actor is created, it doesn't consume any placement group resources.
```

Does the 1 CPU need to be *available* or can the actor still be scheduled even if there is another task/actor using the CPU? If it is the latter, probably need to update my suggestion.

Author: Yea, needs to be available.

Author: Needs to be available.

>   To avoid any surprises, always specify resource requirements explicitly for actors. If resources are specified explicitly, they are required for both scheduling and running.

Reviewer: ```suggestion
  To avoid any surprises, always specify resource requirements explicitly for actors. If resources are specified explicitly, they are required both at schedule time and at execution time.
```

  When you use an actor with a placement group, always specify ``num_cpus``.

>   By default (with no arguments to ``ray.remote``), a Ray actor requires 1 CPU when it is scheduled. But after it is created, it occupies 0 CPUs.

Reviewer: I think this can be clearer. Which of these two options is it?

1. The node where the actor is scheduled has at least 1 CPU, and the CPU is not in use at the time of scheduling.
2. The node where the actor is scheduled has at least 1 CPU, but it could be in use by another task or actor at the time of scheduling.

Also, it'd be useful to say a little bit about why we do this because it sounds a bit strange. I actually didn't know that :D

Author: updated, PTAL.

>   By default (with no arguments to ``ray.remote``) Ray actor requires 1 CPU when it is scheduled. But after it is created, it occupies 0 CPU.

Author: This is the real change

>   By default (with no arguments to ``ray.remote``) Ray actor requires 1 CPU when it is scheduled. But after it is created, it occupies 0 CPU.

Reviewer 2: ```suggestion
  By default (with no arguments to ``ray.remote``), a Ray actor requires 1 CPU when it is scheduled. But after it is created, it occupies 0 CPUs.
```

>   When scheduling an actor without resource requirements and a placement group, the placement group has to be created (since it requires 1 CPU to be scheduled).

Reviewer 2: ```suggestion
  When scheduling an actor with no resource requirements and a placement group, the placement group gets created (since the actor requires 1 CPU to be scheduled).
```

>   When scheduling an actor with no resource requirements and a placement group, the placement group gets created (since the actor requires 1 CPU to be scheduled).

Reviewer: This is also a bit unclear; does the placement group need to specify a bundle with {"CPU": 1} for the actor to be scheduled or is it possible to specify an empty bundle?

>   However, after the actor is created, it doesn't consume the placement group resources.

Reviewer 2: ```suggestion
  However, after the actor is created, it doesn't consume any placement group resources.
```
